fury/ui/elements.py

```diff
@@ -3484,3 +3485,281 @@ def left_button_dragged(self,  i_ren, _obj, element):
+        -------
+        float
+            Final time for the progress slider.
+        """
+        return self._progress_bar.max_value
+
+    @final_time.setter
+    def final_time(self, t):
+        """Set final progress slider time value.
+
+        Parameters
+        ----------
+        t: float
+            Final time for the progress slider.
+        """
+        self._progress_bar.max_value = t
+
+    @property
+    def current_time(self):
+        """Get current time of the progress slider.
+
+        Returns
+        -------
+        float
+            Progress slider current value.
+        """
+        return self._progress_bar.value
+
+    @current_time.setter
+    def current_time(self, t):
+        """Set progress slider value.
+
+        Parameters
+        -------
+        t: float
+            Current time to be set.
+        """
+        self._progress_bar.value = t
+        self.current_time_str = t
+
+    @property
+    def current_time_str(self):
+        """Returns current time as a string.
+
+        Returns
+        -------
+        str
+            Current time formatted as a string in the form:`HH:MM:SS`.
+
+        """
+        return self.time_text.message
+
+    @current_time_str.setter
+    def current_time_str(self, t):
+        """Set time counter.
+
+        Parameters
+        ----------
+        t: float
+            Time to be set in the time_text counter.
+
+        Notes
+        -----
+        This should only be used when the `current_value` is not being set
+        since setting`current_value` automatically sets this property as well.
+        """
+        t = np.clip(t, 0, self.final_time)
+        self.time_text.message = time.strftime('%H:%M:%S', time.gmtime(t))
+
+    @property
+    def speed(self):
+        """Returns current speed.
+
+        Returns
+        -------
+        str
+            Current time formatted as a string in the form:`HH:MM:SS`.
+
+        """
+        return self._speed
+
+    @speed.setter
+    def speed(self, speed):
+        """Set time counter.
+
+        Parameters
+        ----------
+        speed: float
+            Speed value to be set in the speed_text counter.
+        """
+        if speed <= 0:
+            speed = 0.01
+        self._speed = speed
+        speed_str = f"{speed}".strip("0").rstrip('.')
+        self.speed_text.font_size = 21 if .01 <= speed < 100 else 14
+        self.speed_text.message = speed_str
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.panel.actors, self._progress_bar.actors, self.time_text
+
+    def _add_to_scene(self, _scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        _scene : scene
+
+        """
+        self.panel.add_to_scene(_scene)
+        self._progress_bar.add_to_scene(_scene)
+        self.time_text.add_to_scene(_scene)
+
+    def _set_position(self, _coords):
```

Comment:
Maybe raise a NotImplementedError for now, while this is not implemented.

Reply:
since these methods override methods on the UI class, and the UI requires these methods, I just want them to do nothing for now.